### PR TITLE
Mesh 2133/return extrinsic errors

### PR DESCRIPTION
**Pull request by HenriqueNogara**
Calling `call_runtime_with_error` writes a `Result<(), String>` to the output buffer, which can be used for more descriptive error msg.

### new features

 - Add support for `CallRuntimeWithError` chain_extension;

### modified external API

- Remove support for deprecated chain_extension `OldCallRuntime`;

### new external API

 - Add support for `CallRuntimeWithError` chain_extension;

### CODE REVIEW

File: contracts/upgradeable-polymesh-ink/src/lib.rs [hunk 1]
```diff
@@ -8,109 +8,59 @@ mod macros;
...
-    InkDelegateCallError {
-        selector: [u8; 4],
-        err: Option<InkEnvError>,
-    },
-}
-
-/// Encodable `ink::env::Error`.
-#[derive(Debug, scale::Encode, scale::Decode)]
-#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
-pub enum InkEnvError {
```
Comment (HenriqueNogara) [inline contracts/upgradeable-polymesh-ink/src/lib.rs]: This was not being used. Let me know if it's not ok to remove this.

File: contracts/upgradeable-polymesh-ink/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ ink = { version = "4.3", default-features = false }
 scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
 scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
 
-polymesh-api = { version = "3.5", default-features = false, features = ["ink"] }
+polymesh-api = { path = "../../../polymesh-api", default-features = false, features = ["ink"] }
```
Comment (HenriqueNogara) [inline contracts/upgradeable-polymesh-ink/Cargo.toml]: Change this once polymesh-api is merged

File: contracts/upgradeable-polymesh-ink/src/lib.rs (hunk 1 above)
Comment (Neopallium) [reply contracts/upgradeable-polymesh-ink/src/lib.rs]: I just remembered why we had this.  We will need it again.  The delegate calls (see in `src/macros.rs`) can fail.  During the Ink! 3.x -> 4.x upgrade I didn't switch to using `try_invoke` to handle those errors.

So we are going to need this again.

File: contracts/upgradeable-polymesh-ink/Cargo.toml (hunk 2 above)
Comment (Neopallium) [reply contracts/upgradeable-polymesh-ink/Cargo.toml]: This can be updated to version 3.6 now.

**Review by Neopallium — APPROVED** [review]

File: contracts/upgradeable-polymesh-ink/Cargo.toml [hunk 3]
```diff
@@ -15,7 +15,8 @@ ink = { version = "4.3", default-features = false }
 scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
 scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
 
-polymesh-api = { version = "3.5", default-features = false, features = ["ink"] }
+polymesh-api = { version = "3.6", default-features = false, features = ["ink"] }
+polymesh-api-ink = { version = "1.2.2", default-features = false, features = ["use_call_runtime_with_error"] }
```
Comment (Neopallium) [inline contracts/upgradeable-polymesh-ink/Cargo.toml]: We shouldn't enable this by default.  We can add a feature flag to this file to allow turning it on.  After this PR is merged and testnet/mainnet has it included, we can make it the default.

**Review by Neopallium — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**